src/scales/scale.time.js

```diff
@@ -50,19 +50,36 @@ module.exports = function(Chart) {
 		return a - b;
 	}
 
-	function buildLookupTable(ticks, min, max, linear) {
-		var ilen = ticks.length;
+	/**
+	 * Returns an array of {time, pos} objects used to interpolate a specific `time` or position
+	 * (`pos`) on the scale, by searching entries before and after the requested value. Since the
+	 * value is linearly interpolated, only timestamps that break the time linearity are added,
+	 * meaning that in the best case, it contains only 2 objects: {min, 0} and {max, 1}. `pos` is
```

Comment:
Edit: You may be right on the first part, simplify the linear table with min/max. I would keep the linearity check though since it's a great optim that applies to all sources.